src/releasing/quattorpoddoc.py

```diff
@@ -13,278 +16,384 @@
-    LOGGER.debug("Running pod2markdown on %s." % podfile)
+    logger.debug("Running pod2markdown on %s." % podfile)
     output = run_asyncloop("pod2markdown %s" % podfile)
-    LOGGER.debug("writing output to %s." % outputfile)
-    LOGGER.debug(output)
+    logger.debug("writing output to %s." % outputfile)
+    logger.debug(output)
     with open(outputfile, "w") as fih:
         fih.write(output[1])
 
 
-def addintrogreeter(outputloc):
-    """
-    Writes small intro file.
-    """
-    LOGGER.info("Adding introduction page.")
-    with open(os.path.join(outputloc, DOCDIR, "index.md"), "w") as fih:
-        fih.write("This is the documentation for the core set of configuration modules for configuring systems with Quattor.\n")
-
-
-def generatetoc(pods, outputloc, indexname):
+def generatetoc(mdfiles, outputloc, indexname):
     """
     Generates a TOC for the parsed components.
     """
-    LOGGER.info("Generating TOC as %s." % os.path.join(outputloc, indexname))
+    logger.info("Generating TOC as %s." % os.path.join(outputloc, indexname))
 
     with open(os.path.join(outputloc, indexname), "w") as fih:
-        fih.write("site_name: Quattor Configuration Modules (Core)\n\n")
+        fih.write("site_name: Quattor Documentation\n\n")
         fih.write("theme: 'readthedocs'\n\n")
         fih.write("pages:\n")
-        fih.write("- Home: 'index.md'\n")
+        fih.write("- introduction: 'index.md'\n")
 
-        addintrogreeter(outputloc)
+        for subdivision in sorted(mdfiles.keys()):
+            fih.write("- %s:\n" % subdivision)
+            for page in sorted(mdfiles[subdivision]):
+                linkname = page.split(".")[0]
+                linktopage = "%s/%s" % (subdivision, page)
+                writeifexists(outputloc, linktopage, linkname, fih)
+        fih.write("\n")
 
-        fih.write("- Components:\n")
-        for component in sorted(pods):
-            name = component.replace('ncm-', '')
-            linkname = "%s/%s.md" % (SUBDIR, name)
-            fih.write("    - '%s'\n" % linkname)
-            if len(pods[component]) > 1:
-                for pod in sorted(pods[component][1:]):
-                    subname = os.path.splitext(os.path.basename(pod))[0]
-                    linkname = "%s/%s::%s.md" % (SUBDIR, name, subname)
-                    fih.write("    - '%s'\n" % linkname)
+
+def writeifexists(outputloc, linktopage, linkname, fih):
+    """
+    Checks if the MD exists before adding it to the TOC.
+    """
+    if os.path.exists(os.path.join(outputloc, DOCDIR, linktopage)):
+        logger.debug("Adding %s to toc." % linkname)
+        fih.write("    - %s: '%s'\n" % (linkname, linktopage))
+    else:
+        logger.warn("Expected %s but it does not exist. Not adding to toc."
+                    % os.path.join(outputloc, DOCDIR, linktopage))
 
 
-def removemailadresses(mdfiles):
+def removemailadresses(mdfiles, outputloc):
     """
     Removes the email addresses from the markdown files.
```

Comment:
they are in the source .pod .pm files, and as agreed on last workshop, they should be removed from there when we start cleaning them up.